girder_worker/plugins/docker/executor.py

```diff
@@ -43,35 +47,44 @@ def _transform_path(inputs, taskInputs, inputId, tmpDir):
     raise Exception('No task input found with id = ' + inputId)
 
 
-def _expand_args(args, inputs, taskInputs, tmpDir):
+def _expand_args(args, inputs, taskInputs, outputs, tmpDir):
     """
-    Expands arguments to the container execution if they reference input
+    Expands arguments to the container execution if they reference input or output
     data. For example, if an input has id=foo, then a container arg of the form
-    $input{foo} would be expanded to the runtime value of that input. If that
+    ``$input{foo}`` would be expanded to the runtime value of that input. If that
     input is a filepath target, the file path will be transformed into the
     location that it will be available inside the running container.
+
+    Output file paths can be referenced using ``$output{foo}``. That will expand
+    to use the "name" field of the corresponding output binding for this task invocation.
+    TODO(zach) add example of task output and corresponding output binding. This
+    expansion will use the absolute path format with DATA_VOLUME as the base path.
     """
     newArgs = []
-    inputRe = re.compile(r'\$input\{([^}]+)\}')
-    flagRe = re.compile(r'\$flag\{([^}]+)\}')
 
     for arg in args:
         skip = False
-        for inputId in re.findall(inputRe, arg):
+        for inputId in _inputRe.findall(arg):
             if inputId in inputs:
                 transformed = _transform_path(
                     inputs, taskInputs, inputId, tmpDir)
                 arg = arg.replace('$input{%s}' % inputId, str(transformed))
             elif inputId == '_tempdir':
                 arg = arg.replace('$input{_tempdir}', DATA_VOLUME)
```

Comment:
Ah, ok. It seems there is a bit more nuance to `$input{}` so that the code paths cannot parallel each other or use the same code? I see that e.g. `os.path.join(DATA_VOLUME, ...)` is used in both places.